### ISSUE TITLE
Show past hourly forecast

### ISSUE THREAD

**Issue by pdpapper** [breezy-weather/breezy-weather#560]
### Describe your idea

Unlike Geometric Weather, Breezy automatically updates the hourly forecast card to start at the current time even when the forecast data has not been updated. I appreciate the convenience of this, but would still like to be able to check what the farecast data were for earlier hours.

Since the data should still be stored locally it should be possible to allow scrolling back to them even if the trend card initially starts with the current hour.

This would also be more consistent with the daily card, which also starts with the current day even if the forecast hasn't been updated online, but does allow scrolling back.

### Other details

This request may be related to closed issue #301 with different goals.

I am not clear on whether what's written there actually means that the previous hourly data is actually discarded or only not shown.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] My idea is not related to a question covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document, or a feature described in the [Homepage](https://github.com/breezy-weather/breezy-weather/blob/main/docs/HOMEPAGE.md) document.
- [X] I have updated the app to **[version 4.6.11 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
I can confirm we now store hourly forecast back to 00:00; Geometric Weather didn't.

Since it's a forecast, it's not an observation. I wanted to display them, but we have reports of AccuWeather users that it's inaccurate on current hour (because on refresh, hourly forecast for AW starts on next hour and not on current hour so we show past values).

So, it will be done as part of #252 but only on daily forecast view, and there will be a warning it must be taken with a pinch of salt.

**Comment by pdpapper** [comment]
Apparently I misunderstood the suggestion being made in issue #301 and so should not have mentioned it. #301 seems to be about keeping old forecast data AFTER a user has updated a location online. My request here is about displaying (I assume it is already kept) the old hourly forecast when a user HAS NOT updated online. This is already the case for the daily forecasts, but not for hourly. Old hourly forecasts ARE shown in Geometric Weather, though, as long as a location has not been updated.

The fact that these forecasts are often different from observations is my main reason for wanting to see them. That comparison allows an estimate of how good the forecast model is for a location, so that we can have a better idea of how much confidence to put in it. Even to possibly make corrections when our actual location is some distance/elevation from the forecast coordinates.

**Comment by papjul** [comment]
We don't make the difference between forecast that just got refreshed and the one that is no longer available during the refresh. We now store everything back to 00:00.

During the refresh process, we restore missing hourly forecast from database and completes with newer data from API. Once the mix of them is stored again in database, we no longer make the difference between what was just refreshed and what was restored from a previous refresh. And honestly, there is no particular reason to show 3 previous forecast for one source, and none for another (AccuWeather which doesn't have past hours in refresh). So the general rule is the same consistent one for all sources: starts from current hour.

Since we now have data back to yesterday 00:00, we now force the hourly forecast card to start at current hour (for the reasons I explained in my previous message). I won't make the hourly forecast card starts at yesterday 00:00 because this kind of chart is not appropriate for that (as opposite to Geometric Weather, we show more than 24 hours of forecast now, can be up to 16 days! of hourly forecast, which is completely insane for a chart that can only show about 5 hours at a time). #252 will help with that and this new chart will be used in daily forecast details view, as explained in my previous message.

**Comment by pdpapper** [comment]
I'm not sure I understand your explanation since my request is about what is shown when the user has not refreshed at all. You mean the database is actually changed even when there's no refresh?

**Comment by papjul** [comment]
> I'm not sure I understand your explanation since my request is about what is shown when the user has not refreshed at all. You mean the database is actually changed even when there's no refresh?

No, we always have data back to yesterday 00:00 available at all time. Whether you refresh at 03:00, at 11:00, at 14:00 or any other hour, the hours available in the hourly forecast are always the same (the data inside each hour changes with refresh of course), so we don't care about when it was refreshed. We display only current hour and next based on current time for consistency, so if current time changes, displayed hourly forecast will also change without altering any data in database, it's just a display.

If you understand some code, this might help you understand, this is what we call in the hourly forecast card:
https://github.com/breezy-weather/breezy-weather/blob/d1281b63b802d751cf68bf6845d9fd908508a00a/app/src/main/java/org/breezyweather/common/basic/models/weather/Weather.kt#L33-L37

**Comment by papjul** [comment]
Past daily forecast is already available.
Some sources don't provide past forecast so you may need to wait for tomorrow if you just added your location today (Breezy Weather keeps old data back to yesterday).

**Comment by jwenjian** [comment]
ahhh, sorry, yes, I just installed today... 👍
